pandas/tests/extension/decimal/array.py

```diff
@@ -59,14 +59,30 @@ def __init__(self, values, dtype=None, copy=False, context=None):
         values = np.asarray(values, dtype=object)
 
         self._data = values
-        # Some aliases for common attribute names to ensure pandas supports
-        # these
-        self._items = self.data = self._data
-        # those aliases are currently not working due to assumptions
-        # in internal code (GH-20735)
-        # self._values = self.values = self.data
         self._dtype = DecimalDtype(context)
 
+    # Certain pandas function currently refer to these attributes
+    # expecting them to return a numpy array of values. here we
+    # just treat them as synonyms for `self._data`.
```

Comment:
This is not correct, so some historical note (not that you should have known, so no blame!): pandas relied in several places on doing things like `getattr(object, 'values')` to ensure to get the underlying ndarray values of an object if it was not yet an ndarray (eg where the object could have been an Index or Series). That's something we don't really like and are trying to get more structure in (but also for us, the pandas code is huge and complex). But so those attributes we added here are rather to ensure that we are *not* using those wrongly. As from pandas code, we should never directly access the "underlying data" of an EA; all interaction with the EA should go through the official documented interface. 

So those attributes here were added in order to have an EA exercise the tests that has some additional attributes that are not part of the official interface, to make sure they don't accidentally mess up some functionality covered in the tests.

And since `values` and `_values` are still commented out, it seems we were (at the time) not yet sure that it would work, or it probably did not yet work. It would be good (but not here in this PR) to at some point try to fix this.

Reply:
See eg https://github.com/pandas-dev/pandas/issues/20735 as one example of this

Reply:
Ahhh, I see. Not brief, but clear!

Reply:
So, this PR is trying to correct something which was put there as a canary, to catch these bugs?
that still doesn't make sense. If you want to catch accesses to these, this is a bad way to do it.

Why not make them properties which throw when accessed?

Reply:
We want to to ensure pandas does not try to "access" the value, but I agree that "setting" the value can throw an error. So converting to the property instead of attribute is indeed fine.

Reply:
How could the original approach catch reads? OTOH, converting to properties will completely hide the problem, unless they are boobytrapped. Maybe I still don't understand what you were doing.

Reply:
Suggestion: uncomment the `values` and `_values` aliasing, run the decimals tests, and see if any test fail.

Tests will be failing, because pandas is assuming it can do `getattr(obj, 'values'`) on some object and expect that it extracts the data of eg a Series, but not the private data of an EA. See https://github.com/pandas-dev/pandas/issues/20735 for an example.

Reply:
I see. and what about data and _items which weren't commented out?

Reply:
Since the tests are working, pandas does not call those (anymore). Or at least not in a way that gives incorrect results. 
But we added them here to have that tested.

Reply:
I see the failing test due to _values, and why it's difficult to figure out the cases. This PR is kind of mired in that purgatory, so I think I'll just close. Ok?